typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -132,20 +150,107 @@ export class WarpCore {
       );
     } else {
       const searchResult = this.findToken(originName, gasAddressOrDenom);
-      assert(searchResult, `IGP token ${gasAddressOrDenom} is unknown`);
+      assert(searchResult, `Fee token ${gasAddressOrDenom} is unknown`);
       igpToken = searchResult;
     }
 
-    this.logger(`Quoted igp gas payment: ${gasAmount} ${igpToken.symbol}`);
+    this.logger(
+      `Quoted interchain transfer fee: ${gasAmount} ${igpToken.symbol}`,
+    );
     return new TokenAmount(gasAmount, igpToken);
   }
 
-  async getTransferRemoteTxs(
-    originTokenAmount: TokenAmount,
-    destination: ChainNameOrId,
-    sender: Address,
-    recipient: Address,
-  ): Promise<Array<WarpTypedTransaction>> {
+  /**
+   * Simulates a transfer to estimate 'local' gas fees on the origin chain
+   */
+  async getLocalTransferFee({
+    originToken,
+    destination,
+    sender,
+    senderPubKey,
+    interchainFee,
+  }: {
+    originToken: IToken;
+    destination: ChainNameOrId;
+    sender: Address;
+    senderPubKey?: HexString;
+    interchainFee?: TokenAmount;
+  }): Promise<TransactionFeeEstimate> {
+    const originMetadata = this.multiProvider.getChainMetadata(
+      originToken.chainName,
+    );
+    const destinationMetadata =
+      this.multiProvider.getChainMetadata(destination);
+
+    // Check constant quotes first
+    const defaultQuote = this.localFeeConstants.find(
+      (q) =>
+        q.origin === originMetadata.name &&
+        q.destination === destinationMetadata.name,
+    );
+    if (defaultQuote) {
+      return { gasUnits: 0, gasPrice: 0, fee: Number(defaultQuote.amount) };
+    }
+
+    // Form transactions to estimate local gas with
+    const recipient = convertToProtocolAddress(
+      sender,
+      destinationMetadata.protocol,
+      destinationMetadata.bech32Prefix,
+    );
+    const txs = await this.getTransferRemoteTxs({
+      originTokenAmount: originToken.amount(1),
+      destination,
+      sender,
+      recipient,
+      interchainFee,
+    });
+
+    if (txs.length === 1) {
+      return this.multiProvider.estimateTransactionFee({
+        chainNameOrId: originMetadata.name,
+        transaction: txs[0],
+        sender,
+        senderPubKey,
+      });
+    } else if (
+      txs.length === 2 &&
+      originToken.protocol === ProtocolType.Ethereum
+    ) {
+      // For ethereum txs that require >1 tx, we assume the first is an approval
+      // We use a hard-coded const as an estimate for the transferRemote gas
```

Comment:
why hardcode this? seems fragile
imo the warp route interface should be independent from the implementation details
also different ERC20 collaterals have different gas usage

Reply:
How do you suggest we do estimations for transfers that require an approval. Keep in mind we can only rely on the set of operations that free RPCs allow, meaning no fancy multi-tx simulation

Reply:
oh... I did not understand the problem until just now

Reply:
Improving the comments around here to make this more clear